src/web-server.js

```diff
@@ -58,11 +58,13 @@ app.use(auth);
 app.use(express.static(__dirname + "/public"));
 
 function getEnv(req) {
-  if (req.query.env === undefined) {
-    return "default";
-  } else {
-    return req.query.env;
-  }
+  return req.query.env;
+}
+
+function sendError(res, ex, prefix) {
+  const status =
+    ex && ex.message && /No such environment/.test(ex.message) ? 404 : 500;
```

Comment:
Perhaps a bit code-smelly, but this is the logic that determines whether to send a 404 or 500.